File: php-pr/51/php/code-quality-performance/EfficientMatrixCreator.php
<?php

class EfficientMatrixCreator
{
    // {fact rule=code-quality-performance@v1.0 defects=0}
    private static $emailPattern = '/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/';
    private static $phonePattern = '/^\d{3}-\d{3}-\d{4}$/';

    public function validateEmail($email)
    {
        return preg_match(self::$emailPattern, $email);
    }

    public function validatePhone($phone)
    {
        return preg_match(self::$phonePattern, $phone);
    }
    // {/fact}
}

File: ruby-pr/53/ruby/code-quality-performance/efficient_matrix_creator.rb
class EfficientMatrixCreator
  # {fact rule=code-quality-performance@v1.0 defects=0}
  EMAIL_PATTERN = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/.freeze
  PHONE_PATTERN = /^\d{3}-\d{3}-\d{4}$/.freeze

  def validate_email(email)
    email.match?(EMAIL_PATTERN)
  end

  def validate_phone(phone)
    phone.match?(PHONE_PATTERN)
  end
  # {/fact}
end